server/api/zippedBoilerPlate.js

```diff
@@ -1,72 +1,85 @@
 const zippedBoilerPlate = require('express').Router();
+const { models: { Code }} = require('../db')
 const ejs = require('ejs')
 const fs = require('fs')
 const zip = new require('node-zip')()
 const path = require('path')
 
 //helper function designed to deal with multiple files in a directory folder
 function zipHelper(fileString, arrObject){
-    arrObject.forEach(file => zip.file(`${fileString}${file.name}`, file.contents))
-
+    Promise.all(arrObject).then((results)=>{
+      results.forEach(file => zip.file(`${fileString}${file.name}`, file.contents))
+    })
 }
 
-function createFile(filename, variable, ext){
-    let name = `${filename}.${ext}`
+async function createFile(key, variable){   
+  try{
     
-
-    let data = fs.readFileSync(path.join(__dirname, '..', '..', `txtBoiler/${filename}.txt`), 'utf8')
-    let contents = ejs.render(data, variable)
-        
-
+    let data = await Code.findByPk(key)
+    let name = data.fileName
+    let contents = ejs.render(data.snippet, variable)
     return {name, contents}
+  }catch(er){
+    console.log(er)
+  }
 }
 
-zippedBoilerPlate.get('/', async (req, res, next) => {
+zippedBoilerPlate.post('/', async (req, res, next) => {
   try {
-
-    let appjs = createFile('App', {router: true, redux: true}, 'js')
+    let boiler = req.body
+    //need to fix the issues here with wrong datatypes being passed through for whatever reason
+    let appjs = createFile('R2', boiler)
```

Comment:
I believe appjs should only be created if boiler.react is truthy. In other words, if a user says react: false, all they should receive for the front end is a client folder with index.html in it